ecs-cli/modules/cli/local/up_app.go

```diff
@@ -14,18 +14,182 @@
 	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/cli/local/network"
+	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/cli/local/secrets"
+	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/commands/flags"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/secretsmanager"
+	"github.com/aws/aws-sdk-go/service/ssm"
+	composeV3 "github.com/docker/cli/cli/compose/types"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
 	"github.com/urfave/cli"
 )
 
+type ssmDecrypter struct {
+	client *ssm.SSM
+}
+
+type secretsManagerDecrypter struct {
+	client *secretsmanager.SecretsManager
+}
+
+// DecryptSecret returns the decrypted parameter value from SSM.
+func (d *ssmDecrypter) DecryptSecret(arnOrName string) (string, error) {
+	defer func() {
+		// Reset the region of the client in case another SSM secret uses only the param name instead of full ARN.
+		d.client = ssm.New(session.Must(session.NewSessionWithOptions(session.Options{})))
+	}()
+
+	// If the value is an ARN we need to retrieve the parameter name and update the region of the client.
+	paramName := arnOrName
+	if parsedARN, err := arn.Parse(arnOrName); err == nil {
+		resource := strings.Split(parsedARN.Resource, "/") // Resource is formatted as parameter/{paramName}.
+		paramName = strings.Join(resource[1:], "")
+		d.client = ssm.New(session.Must(session.NewSessionWithOptions(session.Options{
+			Config: aws.Config{Region: aws.String(parsedARN.Region)},
+		})))
+	}
+
+	val, err := d.client.GetParameter(&ssm.GetParameterInput{
+		Name:           aws.String(paramName),
+		WithDecryption: aws.Bool(true),
+	})
+	if err != nil {
+		return "", errors.Wrapf(err, "Failed to retrieve decrypted secret from %s due to %v", arnOrName, err)
+	}
+	return *val.Parameter.Value, nil
+}
+
+// DecryptSecret returns the decrypter secret value from Secrets Manager.
+func (d *secretsManagerDecrypter) DecryptSecret(arn string) (string, error) {
+	val, err := d.client.GetSecretValue(&secretsmanager.GetSecretValueInput{
+		SecretId: aws.String(arn),
+	})
+	if err != nil {
+		return "", errors.Wrapf(err, "Failed to retrieve decrypted secret from %s due to %v", arn, err)
+	}
+	return *val.SecretString, nil
+}
+
 // Up creates a Compose file from an ECS task definition and runs it locally.
 //
 // The Amazon ECS Local Endpoints container needs to be running already for any local ECS task to work
 // (see https://github.com/awslabs/amazon-ecs-local-container-endpoints).
 // If the container is not running, this command creates a new network for all local ECS tasks to join
 // and communicate with the Amazon ECS Local Endpoints container.
 func Up(c *cli.Context) {
-	client := docker.NewClient()
-	network.Setup(client)
+	// TODO When we don't provide any flags, Create() should just check if a "./docker-compose.local.yml" already exists.
+	// If so then Create() should do nothing else, otherwise it should error.
+	Create(c)
```

Comment:
For my own understanding -- are we not interpolating the env vars into the docker compose file on Create, only on Up?